ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +268,11 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            LOGGER.warn("XAResource.recover has to be called first with XAResource.TMSTARTRSCAN argument");
+            throw new XAException(XAException.XAER_PROTO);
```

Comment:
By that evidence though I think I would prefer (if possible) to remove the protection and report the NPE. Reporting the outcome to a HEURISTIC is not right. Potentially we could in the future want to run this by some external third party recovery manager.

@gytis without any protection it wouldn't hurt from what I understand? We should never get in the state where TMSTARTRSCAN was not called AFAICT?